analysis/study_definition_ethnicity.py

```diff
@@ -16,45 +16,99 @@
+                    "10": 0.05,
+                    "11": 0.05,
+                    "12": 0.05,
+                    "13": 0.05,
+                    "14": 0.05,
+                    "15": 0.05,
+                    "16": 0.05,
+                }
+            },
+            "incidence": 0.75,
+        },
+    ),
+    ethnicity16=patients.categorised_as(
+        {
+            "Unknown": "DEFAULT",
+            "White - British": "eth='1'",
+            "White - Irish": "eth='2'",
+            "White - Any other White background": "eth='3'",
+            "Mixed - White and Black Caribbean": "eth='4'",
+            "Mixed - White and Black African": "eth='5'",
+            "Mixed - White and Asian": "eth='6'",
+            "Mixed - Any other mixed background": "eth='7'",
+            "Asian or Asian British - Indian": "eth='8'",
+            "Asian or Asian British - Pakistani": "eth='9'",
+            "Asian or Asian British - Bangladeshi": "eth='10'",
+            "Asian or Asian British - Any other Asian background": "eth='11'",
+            "Black or Black British - Caribbean": "eth='12'",
+            "Black or Black British - African": "eth='13'",
+            "Black or Black British - Any other Black background": "eth='14'",
+            "Other Ethnic Groups - Chinese": "eth='15'",
+            "Other Ethnic Groups - Any other ethnic group": "eth='16'",
+        },
+        return_expectations={
+            "category": {
+                "ratios": {
+                    "White - British": 0.1,
+                    "White - Irish": 0.1,
+                    "White - Any other White background": 0.1,
+                    "Mixed - White and Black Caribbean": 0.1,
+                    "Mixed - White and Black African": 0.1,
+                    "Mixed - White and Asian": 0.025,
+                    "Mixed - Any other mixed background": 0.025,
+                    "Asian or Asian British - Indian": 0.05,
+                    "Asian or Asian British - Pakistani": 0.05,
+                    "Asian or Asian British - Bangladeshi": 0.05,
+                    "Asian or Asian British - Any other Asian background": 0.05,
+                    "Black or Black British - Caribbean": 0.05,
+                    "Black or Black British - African": 0.05,
+                    "Black or Black British - Any other Black background": 0.05,
+                    "Other Ethnic Groups - Chinese": 0.05,
+                    "Other Ethnic Groups - Any other ethnic group": 0.05,
+                }
+            },
+            "incidence": 0.8,
+        },
+    ),
+    ethnicity6=patients.categorised_as(
         {
             "Unknown": "DEFAULT",
             "White": "eth='1'",
             "Mixed": "eth='2'",
-            "Asian": "eth='3'",
-            "Black": "eth='4'",
-            "Other": "eth='5'",
+            "Asian or Asian British": "eth='3'",
+            "Black or Black British": "eth='4'",
+            "Chinese or Other Ethnic Groups": "eth='5'",
```

Comment:
For ethnicity6 you care categorizing the eth variable, which is created using the 16 groupings
So right now the ethnicity6 variable will categorize with 
            "White - British": "eth='1'",
            "White - Irish": "eth='2'",
            "White - Any other White background": "eth='3'",
            "Mixed - White and Black Caribbean": "eth='4'",
            "Mixed - White and Black African": "eth='5'", 
            
I think instead you need to extract a new variable patients.with_these_clinical_events(ethnicity6_codes) and then categorize that 

If you were to run flake8 on this file, you might see that ethnicity6_codes is currently unused

Reply:
ohhh yes! thanks! good spot, will fix

Reply:
fixed https://github.com/opensafely/blood-pressure-sro/blob/ce74d0c6ff100251931cf5d435e2d09808681a3d/analysis/study_definition_ethnicity.py#L92-L131